src/back/spv/writer.rs

```diff
@@ -407,7 +407,16 @@ impl Writer {
             function_type,
         ));
 
-        self.write_block(&ir_function.body, ir_module, ir_function, &mut function)?;
+        let id = self.generate_id();
+        let mut block = self.write_block(
+            &mut ir_function.body.to_vec(),
```

Comment:
That seems wrong. We are passing down a mutable reference to a temporary vector that we aren't using anyway?